search_events_app/utils/query_parameters.py

```diff
@@ -30,6 +30,14 @@ class QueryParameters:
             FROM hive.dw.f_ticket_merchandise_purchase
             WHERE is_valid = 'Y'
         ) AS f ON f.event_id = dw_event.event_id
+        INNER JOIN (
```

Comment:
In order to select less columns of the table, and to filter the results of this subquery before retrieving them to the whole base query, I would do this:

SELECT event
FROM hive.eb.ticket_classes
WHERE 
    deleted='n'
    AND end_sales > now()
    AND (start_sales < now() OR start_sales IS NULL)
GROUP BY event
HAVING (SUM(quantity_total) > SUM(quantity_sold) OR SUM(quantity_total) = 0 OR SUM(is_donation) > 0)

Reply:
Resolved